pathd/path_pcep_pcc.c

```diff
@@ -580,50 +608,104 @@ void handle_pcep_lsp_update(struct ctrl_state *ctrl_state,
 			    struct pcc_state *pcc_state,
 			    struct pcep_message *msg)
 {
+	char err[MAX_ERROR_MSG_SIZE] = "";
 	struct path *path;
 	path = pcep_lib_parse_path(msg);
-	specialize_input_path(pcc_state, path);
-	path->update_origin = SRTE_ORIGIN_PCEP;
-	path->originator = pcc_state->originator;
+	lookup_nbkey(pcc_state, path);
+	/* TODO: Investigate if this is safe to do in the controller thread */
+	path_nb_lookup(path);
+	specialize_incoming_path(pcc_state, path);
 	PCEP_DEBUG("%s Received LSP update", pcc_state->tag);
 	PCEP_DEBUG_PATH("%s", format_path(path));
-	pcep_thread_update_path(ctrl_state, pcc_state->id, path);
+
+	if (validate_incoming_path(pcc_state, path, err, sizeof(err)))
+		pcep_thread_update_path(ctrl_state, pcc_state->id, path);
+	else {
+		/* FIXME: Monitor the amount of errors from the PCE and
+		 * possibly disconnect and blacklist */
+		flog_warn(EC_PATH_PCEP_UNSUPPORTED_PCEP_FEATURE,
+			  "Unsupported PCEP protocol feature: %s", err);
+		pcep_free_path(path);
+	}
 }
 
 void handle_pcep_lsp_initiate(struct ctrl_state *ctrl_state,
```

Comment:
Does this function still need the ctrl_state?

Reply:
I don't want to remove it until it is actually implemented, ctrl_state is need for communicating with the controller, like for scheduling a timer for example.

Reply:
Ok, then maybe we should move this functionality with the shorter descriptions to the pceplib, since it doesnt make sense to have it in multiple disparate places.
I believe there are other similar debugging functions that could probably also be moved to the pceplib, but this can be done separately.

Reply:
I am fin with that, there is multiple functions in path_pcep_debug that could go there.

const char *pcep_event_type_name(pcep_event_type event_type);
const char *pcep_message_type_name(enum pcep_message_types pcep_message_type);
const char *pcep_object_class_name(enum pcep_object_classes obj_class);
const char *pcep_object_type_name(enum pcep_object_classes obj_class,
				  enum pcep_object_types obj_type);
const char *pcep_lsp_status_name(enum pcep_lsp_operational_status status);
const char *pcep_tlv_type_name(enum pcep_object_tlv_types tlv_type);
const char *pcep_ro_type_name(enum pcep_ro_subobj_types ro_type);
const char *pcep_nai_type_name(enum pcep_sr_subobj_nai nai_type);
const char *pcep_metric_type_name(enum pcep_metric_types type);
const char *pcep_nopath_tlv_err_code_name(enum pcep_nopath_tlv_err_codes code);

Reply:
@bradyallenjohnson Do you consider this a blocker for merging this feature ?

Reply:
I think this discussion got moved to the wrong file. For now, can you remove the duplicate error message code and call the pceplib function that returns the lengthier description. Then we can merge this patch. Today I can add a patch to pceplib that returns a shorter description.

Reply:
After talking to Sebastien in private, we'll go ahead and merge this as-is. Later (today or tomorrow) we'll add this code to pceplib and remove it from pathd.